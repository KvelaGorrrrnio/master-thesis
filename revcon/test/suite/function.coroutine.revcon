@coroutine

function square(int x, int output) {
  output += x * x
}

coroutine main(int x) {
  local int a 5
  call @square (a, x)
  delocal int a 5
}
