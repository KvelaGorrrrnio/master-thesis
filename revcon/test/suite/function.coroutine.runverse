test/suite/function.coroutine.revcon:
[store]
  main: Ã˜
  x: 0

[stack]
[spawn main main end () | unresume main (x) | despawn main main begin ()] <|> []

