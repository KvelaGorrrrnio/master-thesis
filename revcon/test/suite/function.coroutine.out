test/suite/function.coroutine.revcon:
[store]
  main: Ã˜
  x: 25

[stack]
[spawn main main begin () | resume main (x) | despawn main main end ()] <|> []

